package/src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -82,12 +82,23 @@ public RetryDecision onRequestError(
             ConsistencyLevel consistencyLevel,
             DriverException driverException,
             int retryNumber) {
-
         RetryDecision retryDecision;
 
         try {
             if (driverException instanceof OverloadedException || driverException instanceof WriteFailureException) {
-                retryDecision = retryManyTimesWithBackOffOrThrow(retryNumber);
+                if (this.maxRetryCount == -1 || retryNumber < this.maxRetryCount){
+                    int retryMillis = getRetryAfterMs(driverException.toString());
+                    if (retryMillis == -1) {
+                        retryMillis = (this.maxRetryCount == -1) ?
```

Comment:
why do we do this check here and use fixedBackoffTimeMillis?  Shoudnt we always default to growingBackOffTimeMillis?

Reply:
I think that is what we used to do before. If maxRetryCount == -1, we just backoff with fixed time always.

Reply:
Is it correct to assume that either fixedBackoffTimeMillis or growingBackOffTimeMillis will be present but not both?  If yes, we should add that check in the constructor and update this condition

Reply:
We need to have both. And the constructor enforces that. Am I missing something?